lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -155,21 +157,26 @@ def sort_file_lines!(dir, lang)
         #
         # @param [String] input_dir The directory where the `.lproj` folders to scan are located
         # @param [String] base_lang The base language used as source of truth that all other languages will be compared against
+        # @param [Array<String>] only_langs The list of languages to limit the generation for. Useful to focus only on a couple of issues or just one language
         # @return [Hash<String, String>] A hash of violations, keyed by language code, whose values are the diff output.
         #
         # @note The returned Hash contains keys only for locales with violations. Locales parsed but without any violations found will not appear in the resulting hash.
         #
-        def find_diffs(input_dir:, base_lang:)
+        def find_diffs(input_dir:, base_lang:, only_langs: nil)
           Dir.mktmpdir('a8c-lint-translations-') do |tmpdir|
             # Run SwiftGen 
-            (config_file, langs) = generate_swiftgen_config!(input_dir, tmpdir)
+            langs = only_langs.nil? ? nil : only_langs + [base_lang]
+            (config_file, langs) = generate_swiftgen_config!(input_dir, tmpdir, only_langs: langs)
             Action.sh(swiftgen_bin, 'config', 'run', '--config', config_file)
             
             # Run diffs
             base_file = sort_file_lines!(tmpdir, base_lang)
             langs.delete(base_lang)
             return Hash[langs.map do |lang|
               file = sort_file_lines!(tmpdir, lang)
+              # If the lang ends up not having any translation at all (e.g. a `.lproj` without any `.strings` file in it but maybe just an storyboard or assets catalog), ignore it
+              next nil if file.nil? || File.size(file) <= 10 && File.readlines(file).all? { |line| line.chomp.length == 0 }
```

Comment:
Right, I should add a comment on this. So basically the idea is to avoid any empty file, including a file that only contains blank lines, but not waste any memory by reading a potentially huge file.
So the idea is that the files are usually either completely empty (size zero, actually rare)… or are only containing a couple of `\n` newlines and that's it – because of the original template having newlines to space things up between header, (in this case empty) content, and footer. So my idea here was to only check if the file was small enough before trying to `readlines` and read its whole content to check if it was only a couple of newlines, vs reading the whole files for all cases including when the file is multiple KBs and consuming RAM for nothing.

Definitively needs an explanation comment though I agree 👍

Reply:
I actually ended up thinking of a better and more logical solution to "check if the file is all-empty-lines without reading the whole file if it's too large". Pushed in d50bd33 🙂